### PR TITLE
OCPBUGS-33877: bump aws bootstrap destroy timeout

### PR DESCRIPTION
**Pull request by patrickdillon**
In CI, we are seeing that deleting the SSH rule is frequently hitting the 5 minute timeout. We want to rework how destroying this rule is handled, but in the meantime let's bump the timeout (and add a bit of debug logging) to relieve CI failures.

### CODE REVIEW

**Review by r4f4 — COMMENTED** [review]
/approve

File: pkg/infrastructure/aws/clusterapi/aws.go [hunk 1]
```diff
@@ -250,9 +250,11 @@ func removeSSHRule(ctx context.Context, cl k8sClient.Client, infraID string) err
 		if sg, ok := c.Status.Network.SecurityGroups[capa.SecurityGroupControlPlane]; ok {
```
Comment (sadasu) [inline pkg/infrastructure/aws/clusterapi/aws.go:250]: Is the security group for bootstrap SSH specified here? https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/api/v1beta1/network_types.go#L165

If not, should it be? If yes, should it be deleted from NetworkSpec?

Comment (patrickdillon) [reply pkg/infrastructure/aws/clusterapi/aws.go:250]: No, the SSH rule is added to the Control Plane Security Group (created by CAPA). We don't want to delete the security group, just the rule.

**Comment by patrickdillon** [comment]
/payload-aggregate periodic-ci-openshift-release-master-ci-4.17-e2e-aws-ovn-upgrade 10

File: pkg/infrastructure/aws/clusterapi/aws.go (hunk 1 above)
Comment (patrickdillon) [reply pkg/infrastructure/aws/clusterapi/aws.go:250]: Creating a separate security group for bootstrap is a good idea, but not what we're currently doing.  #8425 will delete the rule without using capi.

**Comment by neisw** [comment]
/payload-aggregate periodic-ci-openshift-release-master-ci-4.17-e2e-aws-ovn 10

**Review by barbacbd — COMMENTED** [review]
/lgtm

**Comment by neisw** [comment]
/hold 
Need to discuss the [revert](https://github.com/openshift/installer/pull/8432) for 4.17 and how this impacts that.

**Comment by neisw** [comment]
/hold cancel
@patrickdillon confirmed no conflicts with the reverts

**Comment by patrickdillon** [comment]
rebased

**Comment by patrickdillon** [comment]
/payload-aggregate periodic-ci-openshift-release-master-ci-4.17-e2e-aws-ovn 10

**Comment by patrickdillon** [comment]
/hold
capi has been reverted so let's just favor the ultimate fix of destroying the rule with the sdk

**Comment by patrickdillon** [comment]
/payload-aggregate periodic-ci-openshift-release-master-ci-4.17-e2e-aws-ovn 10

**Comment by patrickdillon** [comment]
After reviewing #8425 and working on a similar implementation in #8442, I have come full circle and think that increasing the timeout is the best path forward. The implementations in #8425 & #8442 are somewhat complex and basically the exact same implementation that is in CAPA. Re-implementing this logic in the installer is more code for us to maintain and increases the likelihood that we will introduce more bugs.

During our initial investigation of this bug, we thought that running this `Update` on the `AWSCluster` would cause a full reconcile, and potentially cause the bootstrap ignition buckets to be recreated. Observing test runs in this PR, we can see that most runs are able to complete the bootstrap delete in ~10 seconds:

```
time="2024-05-20T19:26:34Z" level=info msg="Waiting up to 10m0s (until 7:36PM UTC) for bootstrap SSH rule to be destroyed..."
...
time="2024-05-20T19:26:44Z" level=debug msg="The bootstrap SSH security rule Bootstrap SSH Access has been removed from sg-0e4815d8d7317a03b"
```
from [here](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/logs/openshift-installer-8427-ci-4.17-e2e-aws-ovn/1792618381476630528/artifacts/e2e-aws-ovn-9/ipi-install-install/artifacts/.openshift_install-1716234234.log)

This [test](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/logs/openshift-installer-8427-ci-4.17-e2e-aws-ovn/1792618370412056576/artifacts/e2e-aws-ovn-3/ipi-install-install/artifacts/.openshift_install-1716234328.log)  captures one of the outliers where it took 8 minutes to fully process the SSH rule. It's unclear why it is taking so long in such a case. Further debugging messages have been added to help identify and potentially debug further occurrences.

Regarding our initial impression that the delete was causing the ignition bucket to be recreated, I do not think that is a correct read of the situation. In one of [the initial runs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_api/1892/pull-ci-openshift-api-master-e2e-aws-serial-techpreview/1791181315274444800/artifacts/e2e-aws-serial-techpreview/ipi-install-install/artifacts/.openshift_install-1715888034.log) where we identified the issue of `created bucket`, the `created bucket` log line appears multiple times before running the command to delete the ssh rule. Reading [the CAPA code](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/s3/s3.go#L256), it appears that CAPA will display the `created bucket` log line even when the bucket already exists. In the example above that took 8 mins, there is only one instance of the `created bucket` log line.

We can definitely revisit whether using the SDK to delete is indeed superior; but it looks like continuing to use CAPA is simpler and does not have the problems we initially thought it did.

**Comment by patrickdillon** [comment]
thanks! will remove the hold when we see positive results from https://github.com/openshift/installer/pull/8427#issuecomment-2123119004

**Comment by patrickdillon** [comment]
/override  ci/prow/e2e-vsphere-ovn
failed two e2es, sandboxes (a common flake) and 

> : [sig-etcd] etcd should not log excessive took too long messages expand_more

which indicates ci environment issues

no sense throwing more jobs at this one

**Pull request merged**